metadata-ingestion/examples/ml/entities/run.py

```diff
@@ -0,0 +1,86 @@
+import argparse
+from typing import List
+import logging
+import time
+import sys
+import datahub.metadata.schema_classes as models
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.ingestion.graph.client import DatahubClientConfig, DataHubGraph
+from datahub.metadata._schema_classes import DataProcessInstancePropertiesClass, AuditStampClass
+from model import Model
+from base_ml_entity import _BaseMLEntity
+from dataset import DataSetModel
+logger = logging.getLogger(__name__)
+class Run(_BaseMLEntity):
+
+    models: List[Model] = []
+    input_datasets:List[DataSetModel] = []
+    output_datasets:List[DataSetModel] = []
+
+    def __init__(self,
+                 run_id: str,
+                 name: str,
+                 client: DataHubGraph):
+        self.run_id = run_id
+        self.name = name
+        self.client = client
+        self.dpi_urn = f"urn:li:dataProcessInstance:{run_id}"
+
+        self.dpi_subtypes = models.SubTypesClass(typeNames=["ML Training Run"])
+
+        # Create the properties aspect
+        self.dpi_props = DataProcessInstancePropertiesClass(
+            name=name,
+            created=AuditStampClass(
+                time=int(time.time() * 1000), actor="urn:li:corpuser:datahub"
+            ),
+        )
+
+        self._emit()
+
+    def _emit(self):
+        mcps = [
+            MetadataChangeProposalWrapper(entityUrn=str(self.dpi_urn), aspect=self.dpi_props),
+            MetadataChangeProposalWrapper(entityUrn=str(self.dpi_urn), aspect=self.dpi_subtypes),
+        ]
+        with client:
+            for mcp in mcps:
+                logger.debug(f"Emitting {mcp}")
+                self.client.emit(mcp)
+
+        logger.info(f"Emitted mcps: {len(mcps)}")
+
+    def add_model(self, model):
```

Comment:
we probably need to add run.add_model_group ()

since it's model group has trainingJobs as well